### PR TITLE
[ALLUXIO-1939] Support CodeStyle Import/Export from IntelliJ

### PR DESCRIPTION
**Pull request by ilganeli**
* Created files that represent IntelliJ CodeStyle configurations
* These files do not need to be edited manually, rather, one should use the "import" and "update" scripts to either import new settings into IntelliJ, or to capture updates made within IntelliJ within Git.

We chose this approach rather than committing a JAR directly because it allows straightforward code review of changes. 

https://alluxio.atlassian.net/browse/ALLUXIO-1939

### CODE REVIEW

**Comment by ilganeli** [comment]
@brennonyork Please review :)

File: conf/CodeStyle/IntelliJ/options/code.style.schemes.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<application>
+  <component name="CodeStyleSchemeSettings">
+    <option name="CURRENT_SCHEME_NAME" value="AlluxioStyleCopy2" />
+  </component>
+</application>
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/options/code.style.schemes.xml]: add missing new line

File: conf/CodeStyle/IntelliJ/update [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: shell scripts should follow the Google shell style guide: https://google.github.io/styleguide/shell.xml

I will inline my comments

File: conf/CodeStyle/IntelliJ/codestyles/AlluxioStyleCopy2.xml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    </indentOptions>
+  </codeStyleSettings>
+  <codeStyleSettings language="XML">
+    <indentOptions>
+      <option name="INDENT_SIZE" value="2" />
+      <option name="CONTINUATION_INDENT_SIZE" value="4" />
+      <option name="TAB_SIZE" value="2" />
+    </indentOptions>
+  </codeStyleSettings>
+</code_scheme>
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/codestyles/AlluxioStyleCopy2.xml]: add missing new line

File: conf/CodeStyle/IntelliJ/import [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+_CALLING_DIR="$(pwd)"
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/import]: unused?

File: conf/CodeStyle/IntelliJ/import [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+
+jar cf AlluxioCodeStyle.jar codestyles/AlluxioStyleCopy2.xml options/code.style.schemes.xml
```
Comment (brennonyork) [inline conf/CodeStyle/IntelliJ/import]: make these absolute paths, use the `-` flag to ensure its read properly across various shell implementations

File: conf/CodeStyle/IntelliJ/update [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+_CALLING_DIR="$(pwd)"
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: `CALLING_DIR`

File: conf/CodeStyle/IntelliJ/update [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+_UNPACK_DIR="$_CALLING_DIR/unpacked"
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: `UNPACK_DIR="${CALLING_DIR}/unpacked"`

File: conf/CodeStyle/IntelliJ/update [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+if [ $# -ne 1 ]; then
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: use `[[ ... ]]` instead of `[ ... ]`

File: conf/CodeStyle/IntelliJ/update [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    echo "Please call with the jar to unpack, e.g. ./update AlluxioCodeStyle.jar" 
+    exit 1
+fi
+
+if [ ! -d "$_UNPACK_DIR" ]; then
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: `[[ ! -d "${UNPACK_DIR}" ]]`

File: conf/CodeStyle/IntelliJ/update [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+  mkdir "$_UNPACK_DIR"
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: `"${UNPACK_DIR}"`

File: conf/CodeStyle/IntelliJ/update [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+fi
+
+
+cd "$_UNPACK_DIR"
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: ditto

File: conf/CodeStyle/IntelliJ/update [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+jar -xvf $_CALLING_DIR/$1
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: "${CALLING_DIR}/$1"

File: conf/CodeStyle/IntelliJ/update [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+cp -rf "$_UNPACK_DIR/codestyles" "$_CALLING_DIR"
+cp -rf "$_UNPACK_DIR/options" "$_CALLING_DIR"
+cd "$_CALLING_DIR"
+rm -rf $"_UNPACK_DIR"
```
Comment (brennonyork) [inline conf/CodeStyle/IntelliJ/update]: this should be `"$_UNPACK_DIR"` not `$"_UNPACK_DIR"` (note where the $ sign is)

File: conf/CodeStyle/IntelliJ/update [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+
+if [ ! -d "$_UNPACK_DIR" ]; then
+  mkdir "$_UNPACK_DIR"
+fi
+
+
+cd "$_UNPACK_DIR"
+
+jar -xvf $_CALLING_DIR/$1
+cp -rf "$_UNPACK_DIR/codestyles" "$_CALLING_DIR"
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: cp -rf "${UNPACK_DIR}/codestyles" "${CALLING_DIR}"

File: conf/CodeStyle/IntelliJ/update [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+cp -rf "$_UNPACK_DIR/options" "$_CALLING_DIR"
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: cp -rf "${UNPACK_DIR}/options" "${CALLING_DIR}"

File: conf/CodeStyle/IntelliJ/update [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+cd "$_CALLING_DIR"
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: `"${CALLING_DIR}"`

File: conf/CodeStyle/IntelliJ/update (hunk 13 above)
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/update]: `"${UNPACK_DIR}"`

File: conf/CodeStyle/IntelliJ/codestyles/AlluxioStyleCopy2.xml [hunk 17]
```diff
@@ -0,0 +1,69 @@
+<code_scheme name="AlluxioStyleCopy2">
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/codestyles/AlluxioStyleCopy2.xml]: can this style be just called `AlluxioStyle` and its file `AlluxioStyle.xml`?

**Comment by jsimsa** [comment]
please add the following header to the newly added xml files

```
<!--
  ~ The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
  ~ (the “License”). You may not use this work except in compliance with the License, which is
  ~ available at www.apache.org/licenses/LICENSE-2.0
  ~
  ~ This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
  ~ either express or implied, as more fully set forth in the License.
  ~
  ~ See the NOTICE file distributed with this work for information regarding copyright ownership.
  -->
```

**Comment by jsimsa** [comment]
that should fix the build failures

**Comment by ilganeli** [comment]
@jsimsa Is there an example license for a bash file?

**Comment by jsimsa** [comment]
I checked and it looks like licenses are not enforced for our shell (which seem wrong) ... but here is code block you can use:

```
# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
# (the “License”). You may not use this work except in compliance with the License, which is
# available at www.apache.org/licenses/LICENSE-2.0
#
# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
# either express or implied, as more fully set forth in the License.
#
# See the NOTICE file distributed with this work for information regarding copyright ownership.
```

File: conf/CodeStyle/IntelliJ/codestyles/AlluxioStyle.xml [hunk 18]
```diff
@@ -66,5 +77,4 @@
       <option name="TAB_SIZE" value="2" />
     </indentOptions>
   </codeStyleSettings>
-</code_scheme>
-
+</code_scheme>
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/codestyles/AlluxioStyle.xml]: add missing new line

**Comment by jsimsa** [comment]
@ilganeli feel free to replace `<exclude>**/conf/*</exclude>` with `<exclude>**/conf/**/*</exclude>` in the top-level pom file, which should silence the build failures you are seeing ... I will investigate the license related build failures independently

**Comment by ilganeli** [comment]
test this please

**Comment by ilganeli** [comment]
retest this please

**Comment by ilganeli** [comment]
@jsimsa What's the keyword to trigger a Jenkins build? I think the issues may be resolved now, I even did a diff against the pom.xml (which has an example license) and things look fine but the check still fails.  If not, I'll add the exclude to the top level config.

**Comment by jsimsa** [comment]
Jenkins, test this please

**Comment by ilganeli** [comment]
Jenkins, test this please

**Comment by jsimsa** [comment]
you might need to be whitelisted to trigger the tests

**Comment by jsimsa** [comment]
Jenkins, test this please

File: conf/CodeStyle/IntelliJ/codestyles/AlluxioStyle.xml [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+    </indentOptions>
+  </codeStyleSettings>
+  <codeStyleSettings language="XML">
+    <indentOptions>
+      <option name="INDENT_SIZE" value="2" />
+      <option name="CONTINUATION_INDENT_SIZE" value="4" />
+      <option name="TAB_SIZE" value="2" />
+    </indentOptions>
+  </codeStyleSettings>
+</code_scheme>
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/codestyles/AlluxioStyle.xml]: missing new line

File: conf/CodeStyle/IntelliJ/options/code.style.schemes.xml [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+  ~ either express or implied, as more fully set forth in the License.
+  ~
+  ~ See the NOTICE file distributed with this work for information regarding copyright ownership.
+  -->
+
+<application>
+  <component name="CodeStyleSchemeSettings">
+    <option name="CURRENT_SCHEME_NAME" value="AlluxioStyle" />
+  </component>
+</application>
```
Comment (jsimsa) [inline conf/CodeStyle/IntelliJ/options/code.style.schemes.xml]: missing new line

**Comment by jsimsa** [comment]
just two minor comments, otherwise LGTM

**Comment by jsimsa** [comment]
@ilganeli thanks for the contribution

**Pull request merged**